Feat/md 7698 kimai improvements

Techbrolakes opened the pull request:
## Description
A clear and concise description of what this pull request adds or changes.

## Types of changes
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

## Checklist
- [ ] I verified that my code applies to the guidelines (`composer code-check`)
- [ ] I updated the documentation (see [here](https://github.com/kimai/www.kimai.org/tree/master/_documentation))
- [ ] I agree that this code is used in Kimai (see [license](https://github.com/kimai/kimai/blob/main/LICENSE))

dnishiyama committed:
Implement docker and customer plugin

dnishiyama committed:
update deployments

dnishiyama committed:
Add customer portal plugin

dnishiyama committed:
Add NotionIntegrationBundle

dnishiyama committed:
clean up plugins

dnishiyama committed:
allow for undefined webhooks envs

Techbrolakes committed:
feat: enhance timesheet form with tag-based description and system improvements

- Added auto-fill description functionality based on selected tags in timesheet form
- Increased Apache request header limits to handle large cookies/tokens (32768 bytes)
- Added template customization support in production Docker build
- Fixed tooltip memory leaks by properly disposing tooltips when elements are removed
- Added new TagAutoFill widget and integrated it with the timesheet form
- Improved code organization

CLAassistant commented:
[![CLA assistant check](https://cla-assistant.io/pull/badge/not_signed)](https://cla-assistant.io/kimai/kimai?pullRequest=5676) <br/>Thank you for your submission! We really appreciate it. Like many open source projects, we ask that you all sign our [Contributor License Agreement](https://cla-assistant.io/kimai/kimai?pullRequest=5676) before we can accept your contribution.<br/>**0** out of **2** committers have signed the CLA.<br/><br/>:x: dnishiyama<br/>:x: Techbrolakes<br/><sub>You have signed the CLA already but the status is still pending? Let us [recheck](https://cla-assistant.io/check/kimai/kimai?pullRequest=5676) it.</sub>

kevinpapst commented:
I assume this was never meant to pop up here.
Anyway: thanks for sharing your code, appreciate it. 
That's the open source spirit and what the AGPL stands for, great work!

Could you please send a few patches upstream (here) in single Pull Requests, like the tooltip fix and whatever you find additionally valuable for the community?
Things like the docker images are internal for you, but certain changes should definitely be shared, so your version is easier to maintain in the long run. 

Also: if you are ever looking for a job in this area, please get in touch!

Techbrolakes committed:
refactor: remove demo data SQL script

- Deleted demo_data.sql script that contained sample data for customers, projects, activities and tags
- Removed test data including Mountain Dev customer, 3 sample projects, 11 activities, and 10 tags
- Script is no longer needed as test data will be managed through application seeding instead
